### PR TITLE
ci: tweak release configuration

### LINKED ISSUE #594: ci: fix invalid release configuration
So, it does create the GitHub release and tags the commit. But for some reason it skips publishing to NPM. There's a bunch of [`Skipped package ... because <older-version> already exists ...` logs](https://github.com/code-pushup/cli/actions/runs/8467225710/job/23197661488#step:7:242), which I don't understand because they reference a different version (`0.29.0`) than what's being correctly calculated (`0.31.0`) in the logs above :confused: 

Also, it seems to commit something and not push it (there's no `release: ...` commit), which causes the subsequent [Code PushUp step to error](https://github.com/code-pushup/cli/actions/runs/8467225710/job/23197661488#step:8:566) (it looks up the current commit with `git log -1`). No clue what's happening there.

_Originally posted by @matejchalk in https://github.com/code-pushup/cli/issues/591#issuecomment-2025729389_

### PR DESCRIPTION
**Pull request by getlarge**
Fixes #594 

---

<details>
<summary><code>migrations.json</code></summary>

```json
{
  "migrations": [
    {
      "cli": "nx",
      "version": "18.0.0-beta.2",
      "description": "Updates nx.json to disabled adding plugins when generating projects in an existing Nx workspace",
      "implementation": "./src/migrations/update-18-0-0/disable-crystal-for-existing-workspaces",
      "x-repair-skip": true,
      "package": "nx",
      "name": "18.0.0-disable-adding-plugins-for-existing-workspaces"
    },
    {
      "version": "18.1.0-beta.3",
      "description": "Moves affected.defaultBase to defaultBase in `nx.json`",
      "implementation": "./src/migrations/update-17-2-0/move-default-base",
      "package": "nx",
      "name": "move-default-base-to-nx-json-root"
    },
    {
      "cli": "nx",
      "version": "18.0.0-beta.0",
      "description": "Add NX_MF_DEV_SERVER_STATIC_REMOTES to inputs for task hashing when '@nx/webpack:webpack' is used for Module Federation.",
      "factory": "./src/migrations/update-18-0-0/add-mf-env-var-to-target-defaults",
      "package": "@nx/react",
      "name": "add-module-federation-env-var-to-target-defaults"
    },
    {
      "cli": "nx",
      "version": "18.1.1-beta.0",
      "description": "Ensure targetDefaults inputs for task hashing when '@nx/webpack:webpack' is used are correct for Module Federation.",
      "factory": "./src/migrations/update-18-1-1/fix-target-defaults-inputs",
      "package": "@nx/react",
      "name": "fix-target-defaults-for-webpack"
    }
  ]
}
```

</details>

### CODE REVIEW

File: global-setup.e2e.ts [hunk 1]
```diff
@@ -7,9 +7,9 @@ import stopLocalRegistry from './tools/scripts/stop-local-registry';
 export async function setup() {
   await globalSetup();
   await startLocalRegistry();
-  execSync('npm install -D @code-pushup/cli@e2e');
-  execSync('npm install -D @code-pushup/eslint-plugin@e2e');
-  execSync('npm install -D @code-pushup/coverage-plugin@e2e');
+  execSync('npm install -D @code-pushup/cli@e2e --force');
+  execSync('npm install -D @code-pushup/eslint-plugin@e2e --force');
+  execSync('npm install -D @code-pushup/coverage-plugin@e2e --force');
```
Comment (getlarge) [inline global-setup.e2e.ts:12]: `force` option was necessary due to the move to the major version update of Nx

File: nx.json [hunk 2]
```diff
@@ -39,8 +39,13 @@
...
     },
     "nx-release-publish": {
-      "dependsOn": ["build"],
-      "executor": "@nx/js:release-publish",
+      "cache": false,
+      "dependsOn": [
+        {
+          "target": "build",
+          "params": "forward"
+        }
```
Comment (getlarge) [inline nx.json:47]: is needed to pass `--skip-nx-cache` flag down to the `nx-release-publish` task pipeline.

File: nx.json [hunk 3]
```diff
@@ -115,5 +117,6 @@
       }
     },
     "releaseTagPattern": "v{version}"
-  }
+  },
+  "useInferencePlugins": false
```
Comment (getlarge) [inline nx.json:121]: automatically added by Nx during migrations. probably to avoid surprises of Project Crystal.

File: global-setup.e2e.ts (hunk 1 above)
Comment (matejchalk) [reply global-setup.e2e.ts:12]: Is it because of `@nx/devkit` peer dependency? We should probably expand `^17.0.0` to `^17.0.0 || ^18.0.0`, so that we're compatible with Nx 18.

File: nx.json (hunk 3 above)
Comment (matejchalk) [reply nx.json:121]: Probably for the best :grin:

File: migrations.json
Comment (matejchalk) [inline migrations.json:1]: The `migrations.json` shouldn't be committed. I added it to the PR description in case we need it for future reference.

File: nx.json [hunk 4]
```diff
@@ -95,16 +100,13 @@
...
         "createRelease": "github"
       }
     },
     "git": {
       "commit": true,
-      "push": true,
-      "pushRemote": "origin",
       "tag": true,
       "commitMessage": "release: {version} [skip ci]"
     },
```
Comment (matejchalk) [inline nx.json:112]: Out of interest, why did you remove these? :thinking:

File: package-lock.json
Comment (matejchalk) [inline package-lock.json:1]: ["`npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing."](https://github.com/code-pushup/cli/actions/runs/8552772054/job/23434588593#step:3:7)

Comment (getlarge) [reply package-lock.json:1]: this smells like messed up node_modules/package-lock.js combination.
I am quite sure that removing the node_modules and package-lock.json and reinstall will solve this.
